app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -63,6 +63,21 @@ def build
         end
       end
 
+      api :PUT, "/hosts/bulk/reassign_hostgroups", N_("Reassign hostgroups")
```

Comment:
REST API wise I think `POST /hostgroups/:id/hosts` would be cleaner but I see the bulk pattern is already established so I'm not going to block on this.